language-design.md

```diff
@@ -27,7 +44,28 @@ How do the themes of _Growing a Language_ relate to the lab we did this week?
 
 **Response**
 
+One way to think about this question is to look at how we grew/changed the language during the lab. We didn't change any of the words (fundamental operations), but we dramatically changed the syntax to allow for more concise use and greater composability. Whereas the original language made composition of sound transformations very challenging, the new syntax allowed for them to be more easily chained.
+
+One approach, which we will refer to as the 'method appraoch' defines a sound class and methods of it which transform and return it, allowing for use like this:
+
+    my_sound.amplify(2).reverse().play()
+   
+One nice property of this approach is that it that it aligns reading/writing order (left to right) with execution order.
+
+Another approach, which we will call the 'function approach' is to have static functions which transformed sound objects like so:
 
+    play(amplify(reverse(my_sound), 2))
+
+This doesn't read/write as nicely, but it allows for the basic functions to be composed, and their compositions would form functions that have the same type as the provided functions.
+
+Zoab notes that even with the method approach users can subclass and add compositions of fundamental operations to their subclass, but acknowledges this does create a difference between user-implemented compositions and the language's fundamental operations. He thinks that aligning read/write ordering with execution ordering is worth creating that distinction.
```

Comment:
Another approach is that used by C++ streams, which allow users to write their own functions which work like the provided `iostream` and `iomanip` ones while maintaining the standard reading order. This also has downsides, though.  It's not very Pythonic (making it harder to learn as a DSL embedded in Python).